### PR TITLE
Added using of 2 proxies

### PR DESCRIPTION
**Pull request by deemerby**

### CODE REVIEW

File: lib/utils/proxy/proxy.go [hunk 1]
```diff
@@ -196,6 +200,77 @@ func dialProxy(ctx context.Context, proxyURL *url.URL, addr string) (net.Conn, e
...
+		Header: make(http.Header),
+	}
+
+	// if cloud proxy has https scheme, need to update connect till tls
+	if cloudProxyURL.Scheme == "https" {
+		tlsConn := tls.Client(conn, &tls.Config{
+			ServerName: cloudProxyURL.Hostname(),
+		})
+
+		conn = tlsConn
```
Comment (vasily-chertkov) [inline lib/utils/proxy/proxy.go:245]: handshake?

Comment (deemerby) [reply lib/utils/proxy/proxy.go:245]: I checked with handshake and without handshake. 
teleport.node was able to connect in the both situations.

I made a decision to remove the handshake

Comment (vasily-chertkov) [reply lib/utils/proxy/proxy.go:245]: It's interesting why it could work without handshake... but ok for now.

**Review by vasily-chertkov — APPROVED** [review]

File: lib/utils/proxy/proxy.go (hunk 1 above)
Comment (deemerby) [reply lib/utils/proxy/proxy.go:245]: yeah, I thought to see the error, but I got 200.

**Pull request merged**